pkg/credentialprovider/plugins.go

```diff
@@ -21,12 +21,23 @@ import (
 	"sort"
 	"sync"
 
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/features"
+
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
 // All registered credential providers.
 var providersMutex sync.Mutex
 var providers = make(map[string]DockerConfigProvider)
+var deprecatedCredentialProviders = sets.NewString(
+	"amazon-ecr",
+	"azure",
+	"google-container-registry",
+	"google-dockercfg",
+	"google-dockercfg-url",
```

Comment:
this seems vulnerable to drift... ideally we could do something in the cloud credential providers themselves that identifies them as such or makes the feature gate effective in skipping them